### PR TITLE
Fix efficiency regression #11436

### LINKED ISSUE #11436: lia performance regression
#### Description of the problem

On 0c86e644 built on shared runner 5, bedrock2 GoFlatToRiscv takes less than 1min [here](https://gitlab.com/coq/coq/-/jobs/406176323#L1637).  
On the next commit, merging #11362 as 55ded808 the same file takes > *15min* ([different runner](https://gitlab.com/coq/coq/-/jobs/407023732#L2236)).
Shortly thereafter (commit 1f3ad4f8), shared runner 5 reports ~ 13min for the same file [here](https://gitlab.com/coq/coq/-/jobs/407271712#L1635).

So I am guessing this is a plain regression in running time, by more than 10x. I will try to find the slow line now.

### PR DESCRIPTION
**Pull request by fajb**
- The cutting plane has been (sometimes) improved to generate stronger
  cuts.
- There is now some support for profiling the Simplex
  see documentation for Show Lia Profile.

**Kind:**  bug fix / performance 

Fixes #11436

- [x] Added  test-suite
- [x] Corresponding documentation was added
- [x] Entry added in the changelog

### CODE REVIEW

File: plugins/micromega/simplex.ml [hunk 1]
```diff
@@ -516,9 +534,7 @@ let cut env rmin sol vm (rst : Restricted.t) tbl (x, v) =
         Vect.null r
     in
     let lcut =
-      List.map
-        (fun cv -> Vect.normalise (cut_vector cv))
-        [cut_coeff1; cut_coeff2]
+      List.map (fun cv -> Vect.normalise (cut_vector cv)) [cut_coeff1]
```
Comment (vbgl) [inline plugins/micromega/simplex.ml]: Using `List.map` on a list with exactly one element seems a bit overkill.

**Review by vbgl — APPROVED** [review]

File: plugins/micromega/simplex.ml (hunk 1 above)
Comment (fajb) [reply plugins/micromega/simplex.ml]: Quite right. There used to be two elements. I'll fix that.

**Comment by vbgl** [comment]
Something went wrong, it seems: there are a few CI failures.

**Comment by fajb** [comment]
Yes, silly me. I look into it.

**Comment by vbgl** [comment]
HoTT failure seems unrelated (also present in current master).

Bench running there: https://ci.inria.fr/coq/job/benchmark-part-of-the-branch/838/console

**Comment by JasonGross** [comment]
It might be nice to have a `Reset Lia Profile` as well, to get accurate profiling info without having to make a fresh session?

**Comment by fajb** [comment]
`Show Lia Profile` consumes and therefore implicitly resets the profile.
It seems enough though an explicit reset is no problem to implement.

**Comment by JasonGross** [comment]
Ah, I think `Show` commands should not modify the document state, and it would be less confusing to split it up into two separate commands.  (This would also bring it more in line with how Ltac Profiling works.)

File: plugins/micromega/simplex.ml [hunk 2]
```diff
@@ -18,6 +18,44 @@ type ('a, 'b) sum = Inl of 'a | Inr of 'b
 
 let debug = false
 
+(** Exploiting profiling information *)
+
+let profile_info = ref []
+let nb_pivot = ref 0
```
Comment (JasonGross) [inline plugins/micromega/simplex.ml]: Is this synchronized with the document state?  I think you might want to use `Summary.ref` here or something like that to make this play nicely with backtracking?

**Comment by fajb** [comment]
@vbgl the bench is quite bad for bedrock. Is there a way to localise the culprit ?

**Comment by fajb** [comment]
I did another try reverting part of the existing strategy.

**Review by Zimmi48 — APPROVED** [review]
Maybe you could add to the doc a mention that this new option is meant as a debug tool (if that's indeed the case)...

**Comment by vbgl** [comment]
https://ci.inria.fr/coq/job/benchmark-part-of-the-branch/839/console

**Comment by fajb** [comment]
Previous commits were all flawed... I need to read math more carefully or only commit Coq extracted code...
 
I hope that I got it right and that there are no regressions.

**Comment by vbgl** [comment]
Latest benchmark result is not as bad as before but there is still a slowdown of about π % on bedrock2.

https://ci.inria.fr/coq/job/benchmark-part-of-the-branch/840/console

**Comment by fajb** [comment]
One last try. This time, I am more optimistic. I have extracted all the bedrock benchmarks and I am doing better. So maybe, this is good enough.

@vbgl another bench ?

**Comment by vbgl** [comment]
https://ci.inria.fr/coq/job/benchmark-part-of-the-branch/841/console

**Comment by vbgl** [comment]
This seems good now! Shall I merge?

**Comment by fajb** [comment]
Just wait a little bit.
I'll do a little bit of cleaning, and rebase...

**Comment by fajb** [comment]
You can merge -- as soon as the CI succeeds...

**Comment by fajb** [comment]
Thanks a lot @vbgl

**Pull request merged**